pkg/pool-gyro/contracts/Gyro2CLPPool.sol

```diff
@@ -158,32 +158,44 @@ contract Gyro2CLPPool is IGyro2CLPPool, BalancerPoolToken {
         uint256 balanceTokenInScaled18,
         uint256 balanceTokenOutScaled18,
         bool tokenInIsToken0,
-        Rounding rounding
+        SwapKind kind
     ) internal view virtual returns (uint256 virtualBalanceIn, uint256 virtualBalanceOut) {
         uint256[] memory balances = new uint256[](2);
         balances[0] = tokenInIsToken0 ? balanceTokenInScaled18 : balanceTokenOutScaled18;
         balances[1] = tokenInIsToken0 ? balanceTokenOutScaled18 : balanceTokenInScaled18;
 
         (uint256 sqrtAlpha, uint256 sqrtBeta) = _getSqrtAlphaAndBeta();
 
-        uint256 currentInvariant = Gyro2CLPMath.calculateInvariant(balances, sqrtAlpha, sqrtBeta, rounding);
-
-        uint256[2] memory virtualOffsets = _calculateVirtualOffsets(currentInvariant, sqrtAlpha, sqrtBeta, rounding);
-
-        virtualBalanceIn = tokenInIsToken0 ? virtualOffsets[0] : virtualOffsets[1];
-        virtualBalanceOut = tokenInIsToken0 ? virtualOffsets[1] : virtualOffsets[0];
-    }
+        // If EXACT_IN, round invariant down to give less tokens to the user. If EXACT_OUT, round invariant up to
+        // charge more tokens from the user.
+        uint256 currentInvariant = Gyro2CLPMath.calculateInvariant(
+            balances,
+            sqrtAlpha,
+            sqrtBeta,
+            kind == SwapKind.EXACT_IN ? Rounding.ROUND_DOWN : Rounding.ROUND_UP
+        );
```

Comment:
The virtual parameters are always proportional to the invariant, and we're rounding one of them up and the other one down/ 

Technically we should send an invariant rounded down to the virtual parameter that we round down, and an invariant rounded up to the parameter that we're rounding up.

I'm not sure it's really worth it though, as you'd need to compute two different invariants. This is only used for swaps, so rounding is not as impactful as with joins / exits.

Reply:
Done, always rounding down now. There's an [issue](https://github.com/balancer/balancer-v3-monorepo/issues/1205) opened so the Gyro team can check this.